netkan/netkan/webhooks/github_mirror.py

```diff
@@ -0,0 +1,315 @@
+        "CPL"               : 'https://opensource.org/licenses/cpl1.0.php',
+        "EFL-1.0"           : 'https://opensource.org/licenses/ver1_eiffel',
+        "EFL-2.0"           : 'https://opensource.org/licenses/EFL-2.0',
+        "Expat"             : 'https://opensource.org/licenses/MIT',
+        "MIT"               : 'https://opensource.org/licenses/MIT',
+        "GPL-1.0"           : 'http://www.gnu.org/licenses/old-licenses/gpl-1.0.en.html',
+        "GPL-2.0"           : 'http://www.gnu.org/licenses/old-licenses/gpl-2.0.en.html',
+        "GPL-3.0"           : 'http://www.gnu.org/licenses/gpl-3.0.en.html',
+        "LGPL-2.0"          : 'https://www.gnu.org/licenses/old-licenses/lgpl-2.0.html',
+        "LGPL-2.1"          : 'https://www.gnu.org/licenses/old-licenses/lgpl-2.1.html',
+        "LGPL-3.0"          : 'http://www.gnu.org/licenses/lgpl-3.0.en.html',
+        "GFDL-1.1"          : 'http://www.gnu.org/licenses/old-licenses/fdl-1.1.en.html',
+        "GFDL-1.2"          : 'http://www.gnu.org/licenses/old-licenses/fdl-1.2.html',
+        "GFDL-1.3"          : 'http://www.gnu.org/licenses/fdl-1.3.en.html',
+        "LPPL-1.0"          : 'https://latex-project.org/lppl/lppl-1-0.html',
+        "LPPL-1.1"          : 'https://latex-project.org/lppl/lppl-1-1.html',
+        "LPPL-1.2"          : 'https://latex-project.org/lppl/lppl-1-2.html',
+        "LPPL-1.3c"         : 'https://latex-project.org/lppl/lppl-1-3c.html',
+        "MPL-1.1"           : 'https://www.mozilla.org/en-US/MPL/1.1/',
+        "Perl"              : 'http://dev.perl.org/licenses/',
+        "Python-2.0"        : 'https://www.python.org/download/releases/2.0/license/',
+        "QPL-1.0"           : 'https://opensource.org/licenses/QPL-1.0',
+        "W3C"               : 'https://www.w3.org/Consortium/Legal/2015/copyright-software-and-document',
+        "Zlib"              : 'http://www.zlib.net/zlib_license.html',
+        "Zope"              : 'http://old.zope.org/Resources/License.1',
+        "Unlicense"         : 'https://unlicense.org/UNLICENSE',
+    }
+
+    def __init__(self, filename):
+        self.filename = Path(filename)
+        self.contents = self.filename.read_text()
+        self._raw     = json.loads(self.contents)
+
+    def can_mirror(self):
+        return (
+            self._raw.get('kind', 'package') == 'package'
+            and self._raw['download_content_type'] in Ckan.mime_to_extension
+            and self.redistributable()
+        )
+
+    def authors(self):
+        auth = self._raw['author']
+        if isinstance(auth, list):
+            return auth
+        else:
+            return [auth]
+
+    def licenses(self):
+        lic = self._raw['license']
+        if isinstance(lic, list):
+            return lic
+        else:
+            return [lic]
+
+    def license_urls(self):
+        return [Ckan.license_urls[lic] for lic in self.licenses() if lic in Ckan.license_urls]
+
```

Comment:
As above regarding `self` + class properties.

Reply:
Is that better? I kind of like having a reminder that a value is a static class property rather than an object property.

Reply:
Hmm, within Python it is unusual to refer to the class in this manner within the object itself. I don't know that it makes any difference in practice. A lot of python stuff is consistency related and while I don't blindly believe that everything needs to be "pythonic" there is value in it being familiar.

For now it's a nitpick, with good test coverage we'll be able to refactor as needed.